programs/bpf_loader/src/lib.rs

```diff
@@ -153,19 +159,25 @@ pub fn load_program_from_bytes(
         deployment_slot,
         programdata,
         account_size,
+        &mut loaded_program_metrics,
     )
     .map_err(|err| {
         ic_logger_msg!(log_collector, "{}", err);
         InstructionError::InvalidAccountData
     })?;
-    load_elf_time.stop();
-    create_executor_metrics.load_elf_us = load_elf_time.as_us();
+    create_executor_metrics.load_elf_us = loaded_program_metrics.load_elf_us;
+    create_executor_metrics.verify_code_us = loaded_program_metrics.verify_code_us;
 
     if use_jit {
```

Comment:
This entire `if use_jit { compile(); }` part should be moved inside `LoadedProgram::new()` because it is going to be used by the new loader as well, so it can be shared.